File: src/test/com/intellij/lang/jsgraphql/frameworks/federation/GraphQLFederationValidationTest.java
package com.intellij.lang.jsgraphql.frameworks.federation;

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase;
import com.intellij.lang.jsgraphql.GraphQLTestUtils;
import com.intellij.lang.jsgraphql.schema.GraphQLSchemaInfo;
import com.intellij.lang.jsgraphql.schema.GraphQLSchemaProvider;
import com.intellij.lang.jsgraphql.schema.library.GraphQLLibraryTypes;
import org.jetbrains.annotations.NotNull;

public class GraphQLFederationValidationTest extends GraphQLTestCaseBase {

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        enableAllInspections();
    }

    @Override
    protected @NotNull String getBasePath() {
        return "/frameworks/federation/validation";
    }

    public void testQueryValidation() {
        GraphQLTestUtils.withLibrary(getProject(), GraphQLLibraryTypes.FEDERATION, () -> {
            doHighlightingTest();

            GraphQLSchemaInfo schemaInfo = GraphQLSchemaProvider.getInstance(getProject()).getSchemaInfo(myFixture.getFile());
            assertNotNull(schemaInfo);
            assertEmpty(schemaInfo.getErrors(getProject()));
        }, getTestRootDisposable());
    }

}

File: tests/testSrc/com/intellij/lang/jsgraphql/frameworks/federation/GraphQLFederationValidationTest.kt
package com.intellij.lang.jsgraphql.frameworks.federation

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase
import com.intellij.lang.jsgraphql.ide.resolve.GraphQLScopeProvider
import com.intellij.lang.jsgraphql.schema.GraphQLSchemaProvider
import com.intellij.lang.jsgraphql.schema.library.GraphQLLibraryTypes
import com.intellij.lang.jsgraphql.withLibrary

class GraphQLFederationValidationTest : GraphQLTestCaseBase() {

  @Throws(Exception::class)
  override fun setUp() {
    super.setUp()
    enableAllInspections()
  }

  override fun getBasePath() = "/frameworks/federation/validation"

  fun testQueryValidation() {
    withLibrary(project, GraphQLLibraryTypes.FEDERATION, {
      doHighlightingTest()

      val schemaInfo = GraphQLSchemaProvider.getInstance(project).getSchemaInfo(myFixture.file)
      assertNotNull(schemaInfo)
      assertEmpty(schemaInfo.getErrors(project))
    }, testRootDisposable)
  }

  fun testEmptySchemaValidation() {
    withLibrary(project, GraphQLLibraryTypes.FEDERATION, {
      val globalScope = GraphQLScopeProvider.getInstance(project).globalScope
      val schemaInfo = GraphQLSchemaProvider.getInstance(project).getSchemaInfo(globalScope)
      assertNotNull(schemaInfo)
      assertEmpty(schemaInfo.getErrors(project))
    }, testRootDisposable)
  }
}
